From: Paul Wise <pabs3@bonedaddy.net>
Date: Wed, 17 Mar 2010 17:44:29 +0700
Subject: Do not report requirements that are available.
Bug-Debian: http://bugs.debian.org/574246
Forwarded: Guillaume Chazarain <guichaz@gmail.com>
--- a/iotop/data.py
+++ b/iotop/data.py
@@ -24,14 +24,13 @@
     has_ctypes = True
 
 if not ioaccounting or not has_ctypes:
-    def boolean2string(boolean):
-        return boolean and 'Found' or 'Not found'
     print 'Could not run iotop as some of the requirements are not met:'
-    print '- Linux >= 2.6.20 with I/O accounting support ' \
-             '(CONFIG_TASKSTATS, CONFIG_TASK_DELAY_ACCT, ' \
-             'CONFIG_TASK_IO_ACCOUNTING):', boolean2string(ioaccounting)
-    print '- Python >= 2.5 or Python 2.4 with the ctypes module:', \
-        boolean2string(has_ctypes)
+    if not ioaccounting:
+        print '- Linux >= 2.6.20 with I/O accounting support ' \
+              '(CONFIG_TASKSTATS, CONFIG_TASK_DELAY_ACCT, ' \
+              'CONFIG_TASK_IO_ACCOUNTING)'
+    if not has_ctypes:
+        print '- Python >= 2.5 or Python 2.4 with the ctypes module'
 
     sys.exit(1)
 
